Core/src/Detector/CylindricalDetectorHelper.cpp

```diff
@@ -0,0 +1,1115 @@
+/// @note this is a strict matching that requires the rotation to be identical
+///
+/// @note throws exception if any of checks fails
+void checkAlignment(
+    const Acts::GeometryContext& gctx,
+    const std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>&
+        volumes) {
+  // Take first transform as reference transform
+  auto refRotation = volumes[0u]->transform(gctx).rotation();
+  // Loop over rest and recursively test
+  for (auto [iv, v] : Acts::enumerate(volumes)) {
+    if (iv > 0) {
+      auto curRotation = v->transform(gctx).rotation();
+      if (not curRotation.isApprox(refRotation)) {
+        std::string message = "CylindricalDetectorHelper: rotation of volume ";
+        message += std::to_string(iv);
+        message += std::string(" is not aligned with previous volume");
+        throw std::invalid_argument(message.c_str());
+      }
+      // Set recursively
+      refRotation = curRotation;
+    }
+  }
+}
+
+/// @brief Helper method to check the volumes in general and throw and excpetion if failes
+///
+/// @param gctx the geometry context
+/// @param volumes the input volumes to be checked
+///
+/// @note throws exception if any of checks fails
+void checkVolumes(
+    const Acts::GeometryContext& gctx,
+    const std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>&
+        volumes) {
+  // A minimal set of checks - exceptions are thrown
+  // - not enough volumes given
+  std::string message = "CylindricalDetectorHelper: ";
+  if (volumes.size() < 2u) {
+    message += std::string("not enough volume given (") +
+               std::to_string(volumes.size());
+    message += std::string(" ), when required >=2.");
+    throw std::invalid_argument(message.c_str());
+  }
+  // - null pointer detected or non-cylindrical volume detected
+  for (auto [iv, v] : Acts::enumerate(volumes)) {
+    // Check for nullptr
+    if (v == nullptr) {
+      message += std::string("nullptr detector instead of volume " +
+                             std::to_string(iv));
+      throw std::invalid_argument(message.c_str());
+    }
+    // Check for cylindrical volume type
+    if (v->volumeBounds().type() != Acts::VolumeBounds::BoundsType::eCylinder) {
+      message +=
+          std::string("non-cylindrical volume bounds detected for volume " +
+                      std::to_string(iv));
+      throw std::invalid_argument(message.c_str());
+    }
+  }
+  // Check the alignment of the volumes
+  checkAlignment(gctx, volumes);
+}
+
+/// @brief Helper method to check the volume bounds
+///
+/// @param gctx the geometry context
+/// @param volumes the input volumes to be checked
+/// @param refCur reference versus current check
+///
+/// @note throws exception if any of checks fails
+void checkBounds(
+    [[maybe_unused]] const Acts::GeometryContext& gctx,
+    const std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>&
+        volumes,
+    const std::vector<std::array<unsigned int, 2u>>& refCur) {
```

Comment:
```suggestion
    const std::vector<std::pair<unsigned int>>& refCur) {
```

Reply:
I will leave the array, I like it, don't see what the pair gains me.